contracts/strategies/allocation/AllocationWithVotingBase.sol

```diff
@@ -0,0 +1,306 @@
+
+        if (payouts.length > 1) {
+            // Transfer funds
+            // Note: this should live in distribution strategy
+            // bool transferSuccess = Transfer.transferPayouts(token, payouts);
+            // if (!transferSuccess) {
+            //     revert Transfer.TRANSFER_FAILED();
+            // }
+
+            // Loop through the payouts and update the votes on each application
+            for (uint256 i; i < payouts.length; i++) {
+                for (uint256 a; a < applicationIds.length; a++) {
+                    if (payouts[i].to == applications[applicationIds[a]].recipient) {
+                        Application storage application;
+                        application = applications[applicationIds[a]];
+                        application.votes = payouts[i].amount;
+                        application.amount += payouts[i].amount;
+                        emit ApplicationVoted(
+                            sender,
+                            applications[applicationIds[a]].applicationId,
+                            payouts[i].amount,
+                            token,
+                            application.recipient
+                        );
+                    }
+                    // No matching application found
+                }
+            }
+
+            return 1;
+        } else {
+            // Transfer funds
+            // Note: this should live in distribution strategy
+            // bool transferSuccess = Transfer.transferPayout(token, payouts[0]);
+            // if (!transferSuccess) {
+            //     revert Transfer.TRANSFER_FAILED();
+            // }
+
+            Application storage application;
+            application = applications[applicationIds[0]];
+            application.votes = payouts[0].amount;
+            application.amount += payouts[0].amount;
+            emit ApplicationVoted(
+                sender, applications[applicationIds[0]].applicationId, payouts[0].amount, token, application.recipient
+            );
+            // Note: Return the total votes for Application?
+            return 1;
+        }
+    }
+
+    /// @notice Generates the payouts for the application(s)
+    function generatePayouts() external payable override onlyOwner returns (bytes memory) {}
+
+    /// ====================================
+    /// ========= Custom Functions =========
+    /// ====================================
+
+    /// @notice Review applications
+    /// @param _data The data to be decoded
+    /// @dev Pool admin can only set applications to `ACCEPTED` or `REJECTED`
+    function reviewApplications(bytes memory _data) external {
+        // Todo: Pool admin can only set applications to `ACCEPTED` or `REJECTED`
+        (bytes32 identityId,,,) = allo.pools(poolId);
+
+        if (!registry.isOwnerOfIdentity(identityId, msg.sender)) {
+            revert UNAUTHORIZED();
+        }
+
+        // Note: How to determine single vs multi application calls?
```

Comment:
we could expect `_data` to always be an array
single review -> array of length 1.
so we could always iterate through the array

Reply:
I was also assuming we would force the `_data` to be an array of `Allocations`.